packages/puppeteer-core/src/cdp/BrowserConnector.ts

```diff
@@ -155,6 +123,40 @@ export async function _connectToCdpBrowser(
   return browser;
 }
 
+/**
+ * Users should never call this directly; it's called when calling
+ * `puppeteer.connect` with `protocol: 'webDriverBiDi'`.
+ *
+ * @internal
+ */
+export async function _connectToBiDiOverCdpBrowser(
+  options: BrowserConnectOptions & ConnectOptions
+): Promise<BidiBrowser> {
+  const {ignoreHTTPSErrors = false, defaultViewport = DEFAULT_VIEWPORT} =
+    options;
+
+  const connection = await getCdpConnection(options);
+
+  const version = await connection.send('Browser.getVersion');
```

Comment:
I think we cannot rely for that on Firefox as it would not have any CDP in the future.

Reply:
This is a BiDi over CDP connector, not a pure BiDi. Implementing `Puppeteer.connect` for pure BiDi is out of scope of this PR